### PR TITLE
Enhanced notifications

### PR DESCRIPTION
**Pull request by qrwteyrutiyoup**
When notifying people were either mentioned or had a message replied to,
the bot is now going to include in the notification any medias that were
in the message, such as images and documents, for instance.

### CODE REVIEW

File: examples/translations/en-us-all.toml [hunk 1]
```diff
@@ -110,10 +110,10 @@ other = "invalid longitude"
 # Notifications messages.
 
 [notification_replied]
-other = "⚠️ *Notification*\n%s replied to a message of yours in *%s*\n\n_%s_"
+other = "⚠️ *Notification*\n[%s](tg://user?id=%d) replied to a message of yours in *%s*\n\n_%s_"
```
Comment (marcopaganini) [inline examples/translations/en-us-all.toml]: Maybe "to your message"?

File: examples/translations/pt-br-all.toml [hunk 2]
```diff
@@ -110,10 +110,10 @@ other = "longitude inválida"
 # Notifications messages.
 
 [notification_replied]
-other = "⚠️ *Notificação*\n%s respondeu uma mensagem sua em *%s*\n\n_%s_"
+other = "⚠️ *Notificação*\n[%s](tg://user?id=%d) respondeu uma mensagem sua em *%s*\n\n_%s_"
```
Comment (marcopaganini) [inline examples/translations/pt-br-all.toml]: Portuguese question (not sure): Respondeu *a* uma mensagem sua or Respondeu uma mensagem sua?

File: util.go [hunk 3]
```diff
@@ -117,3 +117,86 @@ func trDelete(s, substr string) string {
 	}
 	return ret.String()
 }
+
+// createMediaMessage returns a `Chattable' (which can be sent with bot.Send)
```
Comment (marcopaganini) [inline util.go]: Quoting `Chattable' like m4 seems to be confusing godoc (and at least one Go chrome extension I use). Try "Chattable".

File: util.go [hunk 4]
```diff
@@ -117,3 +117,86 @@ func trDelete(s, substr string) string {
...
+// targeted at `destination', made with the media received in the message passed
+// as parameter, if there is any media in this message.
+func createMediaMessage(message *tgbotapi.Message, destination int64, markup *tgbotapi.InlineKeyboardMarkup) (tgbotapi.Chattable, error) {
+	if message == nil {
+		return nil, fmt.Errorf("message is nil")
```
Comment (marcopaganini) [inline util.go]: I'd expand the error message to something like "cannot process empty message" or something more descriptive.

File: util.go [hunk 5]
```diff
@@ -117,3 +117,86 @@ func trDelete(s, substr string) string {
...
+		// Document, video and photo may have caption, up to 200 characters.
+		document.Caption = message.Caption
+		if markup != nil {
+			document.ReplyMarkup = markup
+		}
+		chattable = document
+	case message.Photo != nil:
+		// We get an array with different dimensions of the received photo, and
+		// want to send the largest one. Let's first find out which one it is.
+		largestPhotoDimension := 0
```
Comment (marcopaganini) [inline util.go]: I'd probably break this down into a function, but not sure.

File: util.go [hunk 6]
```diff
@@ -117,3 +117,86 @@ func trDelete(s, substr string) string {
...
+		chattable = location
+	case message.Contact != nil:
+		contact := tgbotapi.NewContact(destination, message.Contact.PhoneNumber, message.Contact.FirstName)
+		if markup != nil {
+			contact.ReplyMarkup = markup
+		}
+		chattable = contact
+	default:
+		// Simple text message or unhandled media type.
+		return nil, fmt.Errorf("no media available")
```
Comment (marcopaganini) [inline util.go]: It appears you're using "err" to signal a condition that is not necessarily an error. Maybe this function should return an "ok" as well?

File: examples/translations/en-us-all.toml (hunk 1 above)
Comment (qrwteyrutiyoup) [reply examples/translations/en-us-all.toml]: Done

File: examples/translations/pt-br-all.toml (hunk 2 above)
Comment (qrwteyrutiyoup) [reply examples/translations/pt-br-all.toml]: Used "a uma" in this case. Seem better.

File: util.go (hunk 3 above)
Comment (qrwteyrutiyoup) [reply util.go]: Done, but we may have quotes like that elsewhere. Maybe we should look into it in another PR.

File: util.go (hunk 4 above)
Comment (qrwteyrutiyoup) [reply util.go]: Done. Used your wording.

File: util.go (hunk 6 above)
Comment (qrwteyrutiyoup) [reply util.go]: Used ok, as suggested.

**Pull request merged**